### PR TITLE
2.x: Remove anonymous inner classes

### PR DESCRIPTION
**Pull request by akarnokd**
This PR removes anonymous inner classes.

Related #5150.

### CODE REVIEW

File: src/main/java/io/reactivex/Scheduler.java [hunk 1]
```diff
@@ -440,4 +431,23 @@ public boolean isDisposed() {
             return disposed;
         }
     }
+
+    private static class DelegationTask implements Runnable {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Scheduler.java]: Please avoid the `private` modifier as it usually generates extra accessor methods.

File: src/main/java/io/reactivex/internal/functions/Functions.java [hunk 2]
```diff
@@ -651,10 +512,229 @@ public int compare(Object o1, Object o2) {
...
+            return v;
+        }
+
+        @Override
+        public String toString() {
+            return "IdentityFunction";
+        }
+    }
+
+    private static class EmptyRunnable implements Runnable {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/functions/Functions.java]: Please avoid private. Also have these classes as `final`.

File: src/main/java/io/reactivex/internal/operators/observable/ObservablePublish.java [hunk 3]
```diff
@@ -294,14 +235,14 @@ boolean add(InnerDisposable<T> producer) {
         void remove(InnerDisposable<T> producer) {
             // the state can change so we do a CAS loop to achieve atomicity
             for (;;) {
-                // let's read the current observers array
+                // let'disposable read the current observers array
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/observable/ObservablePublish.java:238]: Unnecessary change due to Find/Replace

File: src/main/java/io/reactivex/internal/operators/observable/ObservableScalarXMap.java [hunk 4]
```diff
@@ -36,7 +36,7 @@ private ObservableScalarXMap() {
     }
 
     /**
-     * Tries to subscribe to a possibly Callable source's mapped ObservableSource.
+     * Tries to subscribe to a possibly Callable source'disposable mapped ObservableSource.
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/observable/ObservableScalarXMap.java]: unnecessary replace

File: src/main/java/io/reactivex/internal/operators/observable/ObservableScalarXMap.java [hunk 5]
```diff
@@ -120,7 +120,7 @@ private ObservableScalarXMap() {
      * Maps a scalar value to an ObservableSource and subscribes to it.
      *
      * @param <T> the scalar value type
-     * @param <R> the mapped ObservableSource's element type.
+     * @param <R> the mapped ObservableSource'disposable element type.
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/observable/ObservableScalarXMap.java]: Unnecessary replace.

File: src/main/java/io/reactivex/internal/operators/single/SingleDelay.java [hunk 6]
```diff
@@ -39,33 +39,57 @@ protected void subscribeActual(final SingleObserver<? super T> s) {
...
         final SequentialDisposable sd = new SequentialDisposable();
         s.onSubscribe(sd);
-        source.subscribe(new SingleObserver<T>() {
-            @Override
-            public void onSubscribe(Disposable d) {
-                sd.replace(d);
+        source.subscribe(new OnSubscribeObserver(sd, s));
+    }
+
+    private class OnSubscribeObserver implements SingleObserver<T> {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/single/SingleDelay.java:45]: These could be named more aptly, for example, `DelayObserver` so it shows up as `io.reactivex.interna.operators.single.SingleDelay$DelayObserver`.

File: src/test/java/io/reactivex/internal/operators/observable/ObservableFlatMapMaybeTest.java [hunk 7]
```diff
@@ -191,7 +191,7 @@ public void middleError() {
         Observable.fromArray(new String[]{"1","a","2"}).flatMapMaybe(new Function<String, MaybeSource<Integer>>() {
             @Override
             public MaybeSource<Integer> apply(final String s) throws NumberFormatException {
-                //return Single.just(Integer.valueOf(s)); //This works
+                //return Single.just(Integer.valueOf(disposable)); //This works
```
Comment (akarnokd) [inline src/test/java/io/reactivex/internal/operators/observable/ObservableFlatMapMaybeTest.java]: No need to change the unit tests.

**Review by akarnokd — COMMENTED** [review]
Not bad but needs a lot of small cleanups.

File: src/main/java/io/reactivex/Scheduler.java (hunk 1 above)
Comment (SleimanJneidi) [reply src/main/java/io/reactivex/Scheduler.java]: Will do

**Comment by SleimanJneidi** [comment]
Thx @akarnokd.  this is WIP, Will address your issues. I might need your help in naming stuff.

**Comment by akarnokd** [comment]
Thanks for taking care of this but there seems to be something fundamentally broken by your changes. Could you start over from a fresh checkout of RxJava?

**Comment by SleimanJneidi** [comment]
@akarnokd You mean from 1.x branch?

**Comment by akarnokd** [comment]
No, RxJava 2.

**Comment by SleimanJneidi** [comment]
So, you think there is no point for me trying to fix that? And it is easier if I start from scratch?

**Comment by akarnokd** [comment]
I can pinpoint what went wrong so redoing it and verifying frequently that the test pass is easier for you and for review.

**Comment by SleimanJneidi** [comment]
@akarnokd Sure, I will do that.
